### ISSUE TITLE
SpawnTab isn't going to wsl home directory

### ISSUE THREAD

**Issue by WTanardi** [wez/wezterm#5902]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

20240730-082727-56a27e93

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Hi, I'm using WSL and I'm trying to ditch Tmux and use wezterm as my multiplexer, the problem is that when I spawn new tab, it doesn't go to my WSL home directory, but goes to the home directory of the windows machine, I've tried setting the spawntab args to both CurrentPaneDomain and DefaultDomain and it does the same thing
![image](https://github.com/user-attachments/assets/da5d87cf-8fdb-4a08-ac15-c007aea43532)

### Configuration

local wezterm = require("wezterm")
local config = wezterm.config_builder()

config.default_domain = "WSL:Ubuntu"

-- Start maximized
local mux = wezterm.mux
wezterm.on("gui-startup", function(cmd)
	local tab, pane, window = mux.spawn_window(cmd or {})
	window:gui_window():maximize()
end)

-- GPU
local gpus = wezterm.gui.enumerate_gpus()
config.webgpu_preferred_adapter = gpus[1]
config.front_end = "WebGpu"

-- Keymaps
local act = wezterm.action
config.disable_default_key_bindings = true
config.leader = { key = "`", timeout_milliseconds = 1000 }
config.keys = {
	-- Map Ctrl+Bspc to delete word
	{
		key = "Backspace",
		mods = "CTRL",
		action = act.SendKey({
			key = "w",
			mods = "CTRL",
		}),
	},
	-- Remap copy and paste [for halmak]
	{
		key = "c",
		mods = "CTRL",
		action = act({
			PasteFrom = "Clipboard",
		}),
	},
	{
		key = "v",
		mods = "CTRL",
		action = act({
			CopyTo = "Clipboard",
		}),
	},
	-- Tmux mappings
	-- Enable typing of leader key when pressed twice
	{
		key = "`",
		mods = "LEADER",
		action = act.SendKey({
			key = "`",
		}),
	},
	-- `c to create new tab
	{
		key = "c",
		mods = "LEADER",
		action = act({
			SpawnTab = "CurrentPaneDomain",
		}),
	},
	-- `q to close tab
	{
		key = "q",
		mods = "LEADER",
		action = act.CloseCurrentTab({ confirm = false }),
	},
	-- Shift+Left to move to left tab
	{
		key = "LeftArrow",
		mods = "SHIFT",
		action = act({
			ActivateTabRelative = -1,
		}),
	},
	-- Shift+Right to move to right tab
	{
		key = "RightArrow",
		mods = "SHIFT",
		action = act({
			ActivateTabRelative = 1,
		}),
	},
	-- Shift+Right to move to right tab
	{
		key = "d",
		mods = "LEADER",
		action = act.ShowDebugOverlay,
	},
}

-- Appearance
config.enable_tab_bar = true
config.use_fancy_tab_bar = false
config.window_background_opacity = 0.65
config.win32_system_backdrop = "Acrylic"
config.font = wezterm.font("CaskaydiaCove NFM")
config.font_size = 14
config.color_scheme = "Tokyo Night"
config.window_decorations = "RESIZE"

return config

### Expected Behavior

It should open a new tab to my WSL home directory "~" which is also the directory that wezterm booted up to
![image](https://github.com/user-attachments/assets/e2003e9a-fe0f-417e-b910-91b233ffa705)

**Comment by mlwarner** [comment]
I can reproduce the reported bug on Windows when setting the default domain and hitting `ctrl+shift+n` to spawn a new window or `ctrl+shift+t` to spawn a new tab. Startup works as expected in a WSL domain, but spawning new tabs and windows does not.

### What Operating System(s) are you seeing this problem on?
Windows

### WezTerm version
20240730-082727-56a27e93

### Configuration

```
local wezterm = require("wezterm")
local config = wezterm.config_builder()

config.default_domain = "WSL:Ubuntu"

return config
```

**Comment by WTanardi** [comment]
I've been playing around with different terminal shells, and I've found that this problem only occurs in sh, bash, and zsh, when I use fish and nushell this problem is resolved

**Comment by jamesgour** [comment]
I'm also facing this when I open a new tab despite having:
	config.default_domain = "WSL:Ubuntu"
	config.default_cwd = "~"

**Comment by TheRustifyer** [comment]
Yes, can confirm that `new tab` behaviour isn't launching any WSL instance, just spawing a new shell on the default domain

**Comment by bartdorsey** [comment]
This is also happening to me. Does anyone at least have a workaround?

**Comment by jagadishxyz** [comment]
- Yes new tab and pane both open default windows directory instead of  **'WSL:Ubuntu'**

- I am also having key bind shortcut problem with **'English India Language or Keyboard'**, i can open horizontal pane with **'ctrl+shirft+alt+%'** but vertical key bind **(")** don't work, while it work with US English work

![Screenshot 2025-01-03 220700](https://github.com/user-attachments/assets/2ee19248-f2ad-423b-b5bf-881a555f0d68)

**Comment by williamjvisuri** [comment]
Hi, if anyone still has this issue, this is my tab config to spawn a new tab in "~" directory instead of /mnt... on WSL2.

{
    key  = "T",
    mods = "CTRL|SHIFT",
    action = act.SpawnCommandInNewTab {
      domain = { DomainName = 'WSL:Ubuntu' }, 
      cwd    = "~",
    },
  },

Br,
William

**Comment by jchantrell** [comment]
See https://github.com/wezterm/wezterm/issues/5503#issuecomment-2600490028 if you've found your way here looking for a possible solution

**Comment by LiminalityFGC** [comment]
> Hi, if anyone still has this issue, this is my tab config to spawn a new tab in "~" directory instead of /mnt... on WSL2.
> 
> ```
> {
>     key  = "T",
>     mods = "CTRL|SHIFT",
>     action = act.SpawnCommandInNewTab {
>         domain = { DomainName = 'WSL:Ubuntu' }, 
>         cwd    = "~",
>     },
> },
> ```
> 
> Br, William

This saved me alot of time and headaches, thank you.

**Comment by amareno** [comment]
It doesn't work for me if cllick on + on the toolbar. works only with CTRL+SHIFT+T

For the initial startup, this snipped worked

```
if wezterm.target_triple:find("windows") ~= nil then
  config.default_prog = { 'wsl.exe', '-d', 'Ubuntu', '--cd', '~' }
end
```
